business/layer.go

```diff
@@ -23,13 +28,38 @@ type Layer struct {
 // Global clientfactory and prometheus clients.
 var clientFactory kubernetes.ClientFactory
 var prometheusClient prometheus.ClientInterface
+var once sync.Once
+var kialiCache cache.KialiCache
+
+func initKialiCache() {
+	once.Do(func() {
+		if config.Get().KubernetesConfig.CacheEnabled {
+			if cache, err := cache.NewKialiCache(); err != nil {
+				log.Errorf("Error initializing Kiali Cache. Details: %s", err)
+			} else {
+				kialiCache = cache
+			}
+		}
+		if isExcludedWorkloadsEmpty() {
+			excludedWorkloads = make(map[string]bool)
+			for _, w := range config.Get().KubernetesConfig.ExcludeWorkloads {
+				excludedWorkloads[w] = true
+			}
+			setExcludedWorkloads(excludedWorkloads)
+		}
+	})
+}
 
 func GetUnauthenticated() (*Layer, error) {
 	return Get("")
 }
 
 // Get the business.Layer
 func Get(token string) (*Layer, error) {
+	// Kiali Cache will be initialized once at first use of Business layer
+	if kialiCache == nil {
+		initKialiCache()
```

Comment:
Just to confirm - the intention here is that if enabling cache failed the first time, it will never tried again before restart? But then, why is the kialiCache == nil check important? Isn't it repetition to the once.Do() ?

Reply:
Yeah, kialiCache == nil is redundant here.
I will make it more clear, perhaps putting the once.Do() outside of initKialiCache.